client/src/components/side-bar.tsx

```diff
@@ -0,0 +1,614 @@
+    setShareOpen(true);
+  };
+
+  const handleCloseShare = () => {
+    setShareOpen(false);
+  };
+
+  function isExpValid(exp: LessonExpectation): boolean {
+    if (!exp.features) {
+      return true;
+    }
+    return validateExpectationFeatures(exp.features);
+  }
+
+  function handleLaunch() {
+    saveChanges();
+    const host = process.env.TUTOR_ENDPOINT || location.origin;
+    const guest = `&guest=${context.user?.name}`;
+    const path = `${host}/tutor?lesson=${lessonId}&admin=true${guest}`;
+    window.location.href = path;
+  }
+  function isLessonValid(): boolean {
+    if (!lessonUnderEdit.lesson) {
+      return false;
+    }
+    return (
+      !error &&
+      lessonUnderEdit.lesson.arch != OPENAI_CLASSIFIER_ARCHITECTURE &&
+      lessonUnderEdit.lesson?.expectations.every((exp: LessonExpectation) =>
+        isExpValid(exp)
+      )
+    );
+  }
+
+  function handleDiscard() {
+    navigate(`/lessons`);
+  }
+
+  function handleSavePopUp(open: boolean): void {
+    setSavePopUp(open);
+  }
+
+  function handleSaveExit(): void {
+    saveChanges();
+    navigate(`/lessons`);
+  }
+
+  function handleSaveContinue(): void {
+    saveChanges();
+    handleSavePopUp(false);
+  }
+
+  function saveChanges(): void {
+    if (!lessonUnderEdit.lesson) {
+      return;
+    }
+    toast("Saving...");
+    const convertedLesson: Lesson = { ...lessonUnderEdit.lesson };
+    if (!lessonId) {
+      convertedLesson.createdBy = context.user?.id || "";
+    }
+    const origId = lessonId || lessonUnderEdit.lesson?.lessonId;
+    updateLesson(origId, convertedLesson, cookies.accessToken)
+      .then((lesson) => {
+        if (lesson) {
+          setLesson(lesson);
+        }
+
+        if (lessonId !== lesson?.lessonId) {
+          // window.location.href = `/lessons/edit?lessonId=${lesson.lessonId}`;
+          setLessonId(lesson.lessonId);
+          // window.location.reload();
+          navigate("/lessons");
+        }
+        toast("Success!");
+      })
+      .catch((err) => {
+        toast("Failed to save lesson.");
+        console.error(err);
+      });
+  }
+
+  const trainAIButtonColor =
+    trainStatus.state !== TrainState.SUCCESS &&
+    trainStatus.state !== TrainState.FAILURE
+      ? "secondary"
+      : trainStatus.state === TrainState.FAILURE
+      ? "error"
+      : !(
+          trainStatus.info &&
+          trainStatus.info?.expectations &&
+          Array.isArray(trainStatus.info?.expectations) &&
+          trainStatus.info.expectations.length > 0
+        )
+      ? "error"
+      : Math.min(...trainStatus.info?.expectations.map((x) => x.accuracy)) >=
+        0.6
+      ? "success"
+      : Math.min(...trainStatus.info?.expectations.map((x) => x.accuracy)) >=
+        0.4
+      ? "warning"
+      : "error";
+
+  const host = process.env.TUTOR_ENDPOINT || location.origin;
+  const lessonLink = `${host}/tutor?lesson=${lessonId}`;
+
+  let lastTrainedString = "Never";
+  const months = [
```

Comment:
For cleanliness, could move this to a separate constants file, especially useful if a list of the months is ever needed elsewhere.